sdk/storage/azure-storage-blob/azure/storage/blob/_quick_query_helper.py

```diff
@@ -27,25 +31,57 @@ class QuickQueryReader(object):  # pylint: disable=too-many-instance-attributes
 
     def __init__(
         self,
-        client=None,
         name=None,
         container=None,
-        progress_callback=None,
+        errors='strict',
         encoding=None,
-        **kwargs
+        headers=None,
+        response=None
     ):
         self.name = name
         self.container = container
-        self.response_headers = None
-        self.total_bytes = None
+        self.response_headers = headers
+        self.size = None
         self.bytes_processed = 0
-        self._progress_callback = progress_callback
-        self._client = client
+        self._errors = errors
         self._request_options = kwargs
         self._encoding = encoding
+        self._parsed_results = DataFileReader(QuickQueryStreamer(response), DatumReader())
+        self._first_result = self._process_record(next(self._parsed_results))
 
     def __len__(self):
-        return self.total_bytes
+        return self.size
+
+    def _process_record(self, result):
+        self.size = result.get('totalBytes')
+        self.bytes_processed = result.get('bytesScanned')
```

Comment:
self._bytes_processed is the only place user can know the progress now, it's weird to let the user poll blob_query_reader.bytes_processed

Reply:
That's true - however the bytesScanned returned by the service is not really a good indication of progress anyway... if customers have access to the total size and can no the size of the data they are receiving I think that would be sufficient.